src/main/java/frc/robot/subsystems/IntakeSubsystem.java

```diff
@@ -8,23 +8,55 @@
 import com.revrobotics.CANSparkMax;
 
 import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.classes.DebugLog;
+import frc.robot.classes.TimeOfFlightSensor;
 import frc.robot.hardware.MotorController;
 import frc.robot.hardware.MotorController.MotorConfig;
 import edu.wpi.first.networktables.GenericEntry;
 import edu.wpi.first.wpilibj.shuffleboard.Shuffleboard;
 import edu.wpi.first.wpilibj.shuffleboard.ShuffleboardTab;
+import edu.wpi.first.wpilibj.shuffleboard.SimpleWidget;
+import edu.wpi.first.wpilibj.shuffleboard.BuiltInWidgets;
+import java.util.Map;
 
 public class IntakeSubsystem extends SubsystemBase implements AutoCloseable {
+  public enum FlightStates{
+    IDLE,
+    CUBE,
+    CONE,
+    CONE_SCORE
+  }
+
   public static final double kConeIntakeSpeed = -0.75;
   public static final double kConeOuttakeSpeed = 0.75;
   public static final double kCubeIntakeSpeed = 0.55;
   public static final double kCubeOuttakeSpeed = -0.55;
 
+  private final double kConeHoldSpeed = 0.35; // change after testing
+  private final double kCubeHoldSpeed = 0.25; // change after testing
+
+  private TimeOfFlightSensor timeOfFlightSensor;
+
+  private final double mmConeActivationThreshold = 450.0; 
+  private final double mmCubeActivationThreshold = 450.0; 
+  private FlightStates tofState = FlightStates.IDLE;
+
   private CANSparkMax motor;
   private CANSparkMax motor2;
+
   private static ShuffleboardTab matchTab = Shuffleboard.getTab("Match");
   private static GenericEntry intakeEntry = matchTab.add("Intake Speed", 0.0).getEntry();
-  private static GenericEntry intakeMode = matchTab.add("Intake Mode", "Cone Mode").getEntry();
+  private SimpleWidget intakeMode = matchTab.add("Intake Mode", true).withWidget(BuiltInWidgets.kBooleanBox).withProperties(Map.of("colorWhenFalse", "Purple", "colorWhenTrue", "Yellow"));
+  private static GenericEntry currentState = matchTab.add("ToF State", FlightStates.IDLE.toString()).getEntry();
+  private static GenericEntry sensor0Up = matchTab.add("Cone ToF sensor up: ", true).getEntry();
+  private static GenericEntry sensor1Up = matchTab.add("Cube ToF sensor up: ", true).getEntry(); 
+  private static GenericEntry tofToggle = matchTab.add("ToF Enabled", true).withWidget(BuiltInWidgets.kToggleButton).getEntry();
```

Comment:
Do we need a TOF toggle button?
If manual controls always override TOF, we shouldn't need this

Reply:
Couldn't this still be useful as a fallback?

Reply:
Nvm, removed this tab